integrations/io.js

```diff
@@ -122,6 +122,17 @@ module.exports = function ({
     },
     async writeInputFile(file, contents) {
       let filePath = path.resolve(absoluteInputFolder, file)
+
+      // Ensure the parent folder of the file exists
+      if (
+        !(await fs
+          .stat(filePath)
+          .then(() => true)
+          .catch(() => false))
```

Comment:
`existsSync` is a better solution for this, and will avoid calling `catch`. Alternatively, `statSync` has `{ throwIfNoEntry: true }` option to avoid the try/catch as well.

```
import fs from 'node:fs'

if (!fs.existsSync(filePath)
```